client/lib/user/user.js

```diff
@@ -272,6 +272,28 @@ User.prototype.set = function( attributes ) {
 	return changed;
 };
 
+User.prototype.decrementSiteCount = function() {
+	const user = this.get();
+	if ( user ) {
+		this.set( {
+			visible_site_count: user.visible_site_count - 1,
+			site_count: user.site_count - 1,
+		} );
+	}
+	this.fetch();
+};
+
+User.prototype.incrementSiteCount = function() {
+	const user = this.get();
+	if ( user ) {
+		return this.set( {
+			visible_site_count: user.visible_site_count + 1,
+			site_count: user.site_count + 1,
+		} );
+	}
+	this.fetch();
```

Comment:
We are not calling `this.fetch` in the original code for `incrementSiteCount`.

Reply:
I wanted `decrementSiteCount` and `incrementSiteCount` to have the same behavior. As I understand it, it's "change the site counts right away, and refetch the user info from server just to be sure that we are not out-of-sync and have the up-to-date info"